src/test/java/io/appium/java_client/service/local/ServerBuilderTest.java

```diff
@@ -1,311 +1,312 @@
+import static org.hamcrest.Matchers.greaterThan;
+import static org.hamcrest.Matchers.is;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertThat;
+import static org.junit.Assert.assertTrue;
+
+import org.apache.commons.validator.routines.InetAddressValidator;
+import org.junit.After;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.openqa.selenium.remote.DesiredCapabilities;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.OutputStream;
+import java.net.InetAddress;
+import java.net.NetworkInterface;
+import java.nio.file.Path;
+import java.util.Enumeration;
+import java.util.List;
+
+public class ServerBuilderTest {
+
+    /**
+     * It may be impossible to find the path to the instance of appium server due to different circumstance.
+     * So user may use environment variables/system properties to define the full path to the server
+     * main.js that is supposed to be default.
+     */
+    private static final String PATH_TO_APPIUM_NODE_IN_PROPERTIES = getProperty(APPIUM_PATH);
+
+    private static final Path ROOT_TEST_PATH = getDefault().getPath("src")
+            .resolve("test").resolve("java").resolve("io").resolve("appium").resolve("java_client");
+
+    /**
+     * This is the path to the stub main.js file
+     */
+    private static final Path PATH_T0_TEST_MAIN_JS = ROOT_TEST_PATH
+            .resolve("service").resolve("local").resolve("main.js");
+
+    private static String testIP;
+    private AppiumDriverLocalService service;
+    private File testLogFile;
+    private OutputStream stream;
+
+    private static String getLocalIP(NetworkInterface intf) {
+        String result = null;
+        for (Enumeration<InetAddress> enumIpAddr = intf.getInetAddresses(); enumIpAddr
+                .hasMoreElements(); ) {
+            String calculated;
+            InetAddress inetAddress = enumIpAddr.nextElement();
+            if (!inetAddress.isLoopbackAddress()) {
+                InetAddressValidator validator = InetAddressValidator.getInstance();
+                calculated = inetAddress.getHostAddress().toString();
+                if (validator.isValid(calculated)) {
+                    result = calculated;
+                    break;
+                }
+            }
+        }
+        return result;
+    }
+
+    /**
+     * initialization.
+     */
+    @BeforeClass public static void beforeClass() throws Exception {
+        for (Enumeration<NetworkInterface> en = NetworkInterface.getNetworkInterfaces(); en
+                .hasMoreElements(); ) {
+            NetworkInterface intf = en.nextElement();
+            if ((testIP = getLocalIP(intf)) != null) {
+                break;
+            }
+        }
+    }
+
+    @After
+    public void tearDown() throws Exception {
+        ofNullable(service).ifPresent(AppiumDriverLocalService::stop);
+
+        if (stream != null) {
```

Comment:
It throws the checked IO exception that needs to be handled by catch. So I decided to don't change code there because lambda with the catch clause looks not so cool.